sdk/program/src/instruction.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::integer_arithmetic)]
```

Comment:
Why blanket the whole file, could mask an issue if something new is added later

Reply:
[This commit](https://github.com/solana-labs/solana/pull/15825/commits/4c269b40b4c4031e52a612459a070718c270f9aa) just moved the allows from crate-level to module-level.  The goal will be to replace all module-level allows with checked math and/or commented block-level allows